Tools/plugins/MuonPogTreeProducer.cc

```diff
@@ -369,6 +369,7 @@ void MuonPogTreeProducer::analyze (const edm::Event & ev, const edm::EventSetup
     {
       nGoodMuons = fillMuons(muons,vertexes,beamSpot);
     }
+  eventId_.nMuons = nGoodMuons;
```

Comment:
Hi Carlo, yes I think is more clear now. What you want to store is the pt, right? but you wrote with the max-pT between all the fits, this is what confused me.